rfcs/text/0008/threat.yml

```
@@ -0,0 +1,104 @@
+  group: 2
+  description: >
+    These fields contain cyber threat information data.
+  type: group
+  reusable:
+    top_level: true
+  fields:
+    - name: indicator
+      level: extended
+      description: Indicator of Compromise information.
+      type: object
+      fields:
+        - name: first_seen
+          level: extended
+          type: date
+          description: The date and time when intelligence source first reported sighting this indicator.
+          example: "2020-11-05T17:25:47.000Z"
+        - name: last_seen
+          level: extended
+          type: date
+          description: The date and time when intelligence source last reported sighting this indicator.
+          example: "2020-11-05T17:25:47.000Z"
+        - name: sightings
+          level: extended
+          type: long
+          description: Number of times this indicator was observed conducting threat activity.
+          example: 20
+        - name: type
+          level: extended
+          type: keyword
+          description: Type of indicator as represented by Cyber Observable in STIX 2.0
+          example: ipv4-addr, domain-name, email-addr
+        - name: description
+          level: extended
+          type: wildcard
+          description: Describes the type of action conducted by the threat.
+          example: IP x.x.x.x was seen delivering Angler EK
+        - name: scanner_stats
+          level: extended
+          type: long
+          description: Count of AV/EDR vendors that successfully detected malicious file or URL.
+          example: 4
+        - name: matched
+          level: extended
+          type: keyword
+          description: Identifies the field in the source document that matched this indicator.
+          example: file.hash.sha256, source.ip
+        - name: provider
+          level: extended
+          type: keyword
+          description: Identifies the name of the intelligence provider.
+        - name: confidence
+          level: extended
+          type: keyword
+          description: Identifies the confidence rating assigned by the provider using STIX confidence scales (N/H/M/L, 0-10, Admirality, WEP, or DNI).
+          example: High, 10, Confirmed by other sources, Certain, Almost Certain / Nearly Certain
+        - name: module
+          level: extended
+          type: keyword
+          description: Identifies the name of specific module this data is coming from.
+          example: threatintel
+        - name: dataset
+          level: extended
+          type: keyword
+          description: Identifies the name of specific dataset from the intelligence source.
+          example: threatintel.{abusemalware,abuseurl,misp.otx.limo}
+        - name: ip
+          level: extended
+          type: ip
+          description: Identifies a threat indicator as an IP address (irrespective of direction).
+          example: 1.2.3.4
+        - name: domain
+          level: extended
+          type: keyword
```

```suggestion
          type: wildcard
```

Wildcard might be appropriate here. WDYT?

---

In checking out the table on this blog [post](https://www.elastic.co/blog/find-strings-within-strings-faster-with-the-new-elasticsearch-wildcard-field), I think because most indicator searches (to include domains) are term queries with full value match, keyword is faster than wildcard for term searches, storage considerations, and also impacts to popular substrings. 

I think we can leave this as `keyword` for now; what do you think?